MSBuildTasks.UnitTests/BuildTestOutputFilesTest.cs

```diff
@@ -0,0 +1,206 @@
+      var task = new BuildTestOutputFiles { Input = new ITaskItem[] { taskItem } };
+
+      task.Execute();
+
+      Assert.That (task.Output.Single().GetMetadata ("Platform"), Is.EqualTo ("x86"));
+    }
+
+    [Test]
+    public void ValidConfiguration_CorrectDockerConfiguration ()
+    {
+      var taskItem = new TaskItem ("MyTest.dll");
+      taskItem.SetMetadata ("TestingConfiguration", "Chrome+NoDb+x86+dockerNet45+release");
+      var task = new BuildTestOutputFiles { Input = new ITaskItem[] { taskItem } };
+
+      task.Execute();
+
+      Assert.That (task.Output.Single().GetMetadata ("ExecutionRuntime"), Is.EqualTo ("dockerNet45"));
+    }
+
+    [Test]
+    public void ValidConfiguration_CorrectBuildConfiguration ()
+    {
+      var taskItem = new TaskItem ("MyTest.dll");
+      taskItem.SetMetadata ("TestingConfiguration", "Chrome+NoDb+x86+dockerNet45+release");
+      var task = new BuildTestOutputFiles { Input = new ITaskItem[] { taskItem } };
+
+      task.Execute();
+
+      Assert.That (task.Output.Single().GetMetadata ("ConfigurationID"), Is.EqualTo ("release"));
+    }
+
+    [Test]
+    public void MultipleConfigurations_CorrectParsing ()
+    {
+      var taskItem = new TaskItem ("MyTest.dll");
+      taskItem.SetMetadata ("TestingConfiguration", "Chrome+NoDb+x86+dockerNet45+release;Firefox+SqlServer2012+x64+dockerNet45+debug");
+      var task = new BuildTestOutputFiles { Input = new ITaskItem[] { taskItem } };
+
+      task.Execute();
+
+      Assert.That (task.Output[1].GetMetadata ("Browser"), Is.EqualTo ("Firefox"));
+      Assert.That (task.Output[1].GetMetadata ("DatabaseSystem"), Is.EqualTo ("SqlServer2012"));
+      Assert.That (task.Output[1].GetMetadata ("Platform"), Is.EqualTo ("x64"));
+      Assert.That (task.Output[1].GetMetadata ("ExecutionRuntime"), Is.EqualTo ("dockerNet45"));
+      Assert.That (task.Output[1].GetMetadata ("ConfigurationID"), Is.EqualTo ("debug"));
+    }
+
+    [Test]
+    public void ValidConfiguration_CopiesItemSpec ()
+    {
+      var taskItem = new TaskItem ("MyTest.dll");
+      taskItem.SetMetadata ("TestingConfiguration", "Chrome+NoDb+x86+dockerNet45+release");
+      var task = new BuildTestOutputFiles { Input = new ITaskItem[] { taskItem } };
+
+      task.Execute();
+
+      Assert.That (task.Output.Single().ItemSpec, Is.EqualTo ("MyTest.dll"));
+    }
+
+    [Test]
+    public void ValidConfiguration_CopiesMultipleIdentifiers ()
+    {
+      const string itemSpec = "MyTest.dll";
+      var taskItem = new TaskItem (itemSpec);
+      taskItem.SetMetadata ("TestingConfiguration", "Chrome+NoDb+x86+dockerNet45+release;Firefox+SqlServer2012+x64+dockerNet45+debug");
+      var task = new BuildTestOutputFiles { Input = new ITaskItem[] { taskItem } };
+
+      task.Execute();
+
+      Assert.That (task.Output[1].ItemSpec, Is.EqualTo (itemSpec));
+    }
+
+    [Test]
+    public void ValidConfiguration_HasUniqueConfigurationID ()
+    {
+      const string itemSpec = "MyTest.dll";
```

Comment:
I adapted the usages in the following way: If we ItemSpec is needed twice, it's a variable, otherwise it's inlined. I suspect that's what you meant, because in this specific case the ItemSpec is needed in the assert and has therefore not been adjusted.

Reply:
Perfect! Sorry for overseeing the second usage. I probably saw that in one test and then commented on a different one. Perfect interpretation of my inperfect comment!